src/CommunityToolkit.Maui/ImageSources/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,254 @@
+﻿namespace CommunityToolkit.Maui.ImageSources;
+
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Maui.Controls;
+
+/// <summary>Gravatar image source.</summary>
+/// <remarks>Note that <see cref="UriImageSource"/> is sealed and can't be used as a parent!</remarks>
+public class GravatarImageSource : StreamImageSource
+{
+	/// <summary>The backing store for the <see cref="CacheValidity" /> bindable property.</summary>
+	public static readonly BindableProperty CacheValidityProperty = BindableProperty.Create(nameof(CacheValidity), typeof(TimeSpan), typeof(GravatarImageSource), TimeSpan.FromDays(1));
+
+	/// <summary>The backing store for the <see cref="CachingEnabled" /> bindable property.</summary>
+	public static readonly BindableProperty CachingEnabledProperty = BindableProperty.Create(nameof(CachingEnabled), typeof(bool), typeof(GravatarImageSource), true);
+
+	/// <summary>The backing store for the <see cref="Email" /> bindable property.</summary>
+	public static readonly BindableProperty EmailProperty = BindableProperty.Create(nameof(Email), typeof(string), typeof(GravatarImageSource), defaultValue: null, propertyChanged: OnEmailPropertyChanged);
+
+	/// <summary>The backing store for the <see cref="Image" /> bindable property.</summary>
+	public static readonly BindableProperty ImageProperty = BindableProperty.Create(nameof(Image), typeof(DefaultImage), typeof(GravatarImageSource), defaultValue: DefaultImage.MysteryPerson, propertyChanged: OnDefaultImagePropertyChanged);
+
+	/// <summary>The backing store for the <see cref="ParentHeight" /> bindable property.</summary>
+	internal static readonly BindableProperty parentHeightProperty = BindableProperty.Create(nameof(ParentHeight), typeof(int), typeof(GravatarImageSource), defaultValue: defaultSize, propertyChanged: OnSizePropertyChanged);
```

Comment:
can we follow the same nameing pattern as the other BP? 
```suggestion
	internal static readonly BindableProperty ParentHeightProperty = BindableProperty.Create(nameof(ParentHeight), typeof(int), typeof(GravatarImageSource), defaultValue: defaultSize, propertyChanged: OnSizePropertyChanged);
```

Reply:
@pictos  The properties are `internal` and therefore the naming rule requires it to begin with a lower case character.

Error	IDE1006	Naming rule violation: The first word, 'ParentHeightProperty', must begin with a lower case character	
Error	IDE1006	Naming rule violation: The first word, 'ParentWidthProperty', must begin with a lower case character

Reply:
I am ok with us changing that rule to make `internal` fields be Pascal cased. @pictos  thoughts?